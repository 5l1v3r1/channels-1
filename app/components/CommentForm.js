// @flow
import React from 'react';
import styles from './CommentForm.scss';

const CommentForm = ({
  description,
  isPosting,
  handleChange,
  handleSubmit,
  handleReset
}: {
  description: string,
  isPosting: boolean,
  handleChange: (event: any) => void,
  handleSubmit: () => void,
  handleReset: () => void
}) => (
  <form className={styles.commentForm} onSubmit={handleSubmit}>
    <textarea
      className={styles.textarea}
      placeholder="Add a public comment..."
      value={description}
      onChange={handleChange}
    />
    <div className={styles.btns}>
      <button
        className={styles.cancelBtn}
        type="reset"
        value="Cancel"
        disabled={isPosting || !description}
        onClick={handleReset}
      >
        Cancel
      </button>
      <button
        className={styles.submitBtn}
        type="submit"
        value="Comment"
        disabled={isPosting || !description}
      >
        Comment
      </button>
    </div>
  </form>
);

export default CommentForm;
