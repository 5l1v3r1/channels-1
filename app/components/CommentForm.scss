@import '../app.global.scss';

.commentForm {
  width: 100%;
  padding: 11px 0;
}

.textarea {
  width: 100%;
  height: 60px;
  margin-bottom: 1rem;
  padding: 0.25rem 0.5rem;
  border-radius: 3px;
  color: $white;
  border-color: $gray;
  background-color: transparent;
  resize: none;

  &:focus {
    outline: 0;
    border-color: $white;
  }
}

.btns {
  display: flex;
  justify-content: flex-end;
}

.cancelBtn,
.submitBtn {
  width: 98px;
  height: 28px;
  padding: 0;
  border: none;
  border-radius: 3px;
  color: $white;
  font-weight: 700;
  line-height: 2rem;
  text-align: center;
  background-color: $gray;
  cursor: default;

  @include font-size(12);

  &:disabled {
    opacity: 0.6;
  }

  &:focus {
    outline: 0;
  }
}

.cancelBtn {
  margin-right: 1rem;
}
