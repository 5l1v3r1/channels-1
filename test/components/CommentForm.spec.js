import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CommentForm from '../../app/components/CommentForm';

function setup() {
  const props = {
    description: '',
    isPosting: false,
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    handleReset: jest.fn()
  };

  const enzymeWrapper = shallow(<CommentForm {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('CommentForm', () => {
  it('should render self with no value', () => {
    const { enzymeWrapper } = setup();
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should render self with value', () => {
    const { enzymeWrapper } = setup();
    enzymeWrapper.setProps({
      description: 'Great video!'
    });
    const tree = toJson(enzymeWrapper);

    expect(tree).toMatchSnapshot();
  });

  it('should call handleChange', () => {
    const { props, enzymeWrapper } = setup();
    const textarea = enzymeWrapper.find('.textarea');

    textarea.simulate('change');

    expect(props.handleChange).toBeCalled();
  });

  it('should call handleReset', () => {
    const { props, enzymeWrapper } = setup();
    const cancelBtn = enzymeWrapper.find('.cancelBtn');

    cancelBtn.simulate('click');

    expect(props.handleReset).toBeCalled();
  });
});
