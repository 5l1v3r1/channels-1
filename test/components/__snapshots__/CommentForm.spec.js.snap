// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentForm should render self with no value 1`] = `
<form
  className="commentForm"
  onSubmit={[Function]}
>
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a public comment..."
    value=""
  />
  <div
    className="btns"
  >
    <button
      className="cancelBtn"
      disabled={true}
      onClick={[Function]}
      type="reset"
      value="Cancel"
    >
      Cancel
    </button>
    <button
      className="submitBtn"
      disabled={true}
      type="submit"
      value="Comment"
    >
      Comment
    </button>
  </div>
</form>
`;

exports[`CommentForm should render self with value 1`] = `
<form
  className="commentForm"
  onSubmit={[Function]}
>
  <textarea
    className="textarea"
    onChange={[Function]}
    placeholder="Add a public comment..."
    value="Great video!"
  />
  <div
    className="btns"
  >
    <button
      className="cancelBtn"
      disabled={false}
      onClick={[Function]}
      type="reset"
      value="Cancel"
    >
      Cancel
    </button>
    <button
      className="submitBtn"
      disabled={false}
      type="submit"
      value="Comment"
    >
      Comment
    </button>
  </div>
</form>
`;
